docs/plugin_developers_guide/07_rules_plugins.rst

```diff
@@ -0,0 +1,432 @@
+   }
+
+The above will cause the *plugin_eval* call to be called if the interval expires or if any readings for the asset *status* arrive. The alternate below will only be called at the defined interval. The data will still contain the buffered readings of the asset *status*, but will call *plugin_eval* every 1000 milliseconds.
+
+.. code-block:: JSON
+
+   {
+       "triggers" : [
+           { "asset" : "status" },
+           { "interval" : 1000, "evaluate" : "interval" }
+           ]
+   }
+
+The code for the Moving Average rule plugin's *plugin_trigger* entry point is shown below.
+
+.. code-block:: C
+
+    /**
+     * Return triggers JSON document
+     *
+     * @return	JSON string
+     */
+    string plugin_triggers(PLUGIN_HANDLE handle)
+    {
+            string ret;
+            AverageRule *rule = (AverageRule *)handle;
+
+            if (!rule)
+            {
+                    ret = "{\"triggers\" : []}";
+                    return ret;
+            }
+
+            // Configuration fetch is protected by a lock
+            rule->lockConfig();
+
+            if (!rule->hasTriggers())
+            {
+                    rule->unlockConfig();
+                    ret = "{\"triggers\" : []}";
+                    return ret;
+            }
+
+            ret = "{\"triggers\" : [ ";
+            std::map<std::string, RuleTrigger *> triggers = rule->getTriggers();
+            for (auto it = triggers.begin();
+                      it != triggers.end();
+                      ++it)
+            {
+                    string source = rule->getSource();
+                    if (source.compare("Readings") == 0)
+                            ret += "{ \"asset\"  : \"" + (*it).first + "\"";
+                    else if (source.compare("Statistics") == 0)
+                            ret += "{ \"statistic\"  : \"" + (*it).first + "\"";
+                    else if (source.compare("Statistics History") == 0)
+                            ret += "{ \"statisticRate\"  : \"" + (*it).first + "\"";
+                    else
+                    {
+                            ret += "{ ";	// Keep JSON valid
+                            Logger::getLogger()->error("Unsupported data source %s, rule will not subscribe to any data", source.c_str());
+                    }
+                    ret += " }";
+                    
+                    if (std::next(it, 1) != triggers.end())
+                    {
+                            ret += ", ";
+                    }
+            }
+
+            ret += " ] }";
+
+            // Release lock
+            rule->unlockConfig();
+
+            return ret;
+    }
+
+Plugin Evaluation
+~~~~~~~~~~~~~~~~~
+
+The *plugin_eval* API call is called with the plugin handle and the data, as a string, which holds the values to be evaluated. The return value is a boolean that is the result of the evaluation. True is returned if the rules conditions are met, otherwise the entry point will return false.
```

Comment:
> API call is called with the 

API function is invoked using the 

The output is a boolean representing the outcome of the evaluation. A value of true is returned if the conditions of the rules are satisfied; otherwise, the entry point will yield false.